app/src/main/java/com/alexbar/layoutbasic/utils/MoviesData.kt

```diff
@@ -0,0 +1,63 @@
+package com.alexbar.layoutbasic.utils
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import com.google.gson.reflect.TypeToken
+import java.io.IOException
+
+fun getJsonDataFromAsset(
+    context: Context,
+    fileName: String
+): String? {
+    val jsonString: String
+    try {
+        jsonString = context.assets.open(fileName).bufferedReader().use {
+            it.readText()
+        }
+    } catch (exp: IOException) {
+        exp.printStackTrace()
+        return null
+    }
+
+    return jsonString
+}
+
```

Comment:
and also here , you can combine the movieList and tvSeriesList functions into a single extension function on the Context class. like :
```
inline fun <reified T> Context.getListFromJsonAsset(fileName: String): MutableList<T>? {
    val jsonFileString: String
    try {
        jsonFileString = assets.open(fileName).bufferedReader().use {
            it.readText()
        }
    } catch (exp: IOException) {
        exp.printStackTrace()
        return null
    }

    val type = object : TypeToken<List<T>>() {}.type
    return Gson().fromJson(jsonFileString, type)
}
```

and you can use it like :

```
val movieList: MutableList<Movie>? = context.getListFromJsonAsset("movies.json")
val seriesList: MutableList<TVSeries>? = context.getListFromJsonAsset("series.json")
```
The getListFromJsonAsset function is an extension function on the Context class in Kotlin, defined with a generic type T. It reads the content of your JSON file. It catches any IOException that might occur during file reading, prints the stack trace, and returns null. 
Then, it creates a TypeToken object representing a list of type T, necessary for Gson deserialization. 
Finally, it deserializes the JSON string into a list of objects of type T using Gson's fromJson method and returns it. This concise function provides a flexible way to deserialize JSON files into lists of various types within Android applications.